pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -142,6 +139,29 @@ func (s *Service) Handle(msg dispatcher.DIDCommMsg) error {
 	return s.handle(&message{Msg: msg, ThreadID: thid, NextStateName: next.Name()})
 }
 
+func (s *Service) validateState(thid string, next state) error {
+	// TODO make it configurable #414
+	// for now we use defaults
+	r := retry.WithDefaults()
+	var current state
+	var err error
+	first := true
+	for first || r.Required() {
+		first = false
+		current, err = s.currentState(thid)
+		if err != nil {
+			return err
+		}
+		logger.Infof("current state : %s", current.Name())
+		if current.CanTransitionTo(next) {
+			return nil
+		}
+		logger.Infof("invalid state transition: %s -> %s [%d] retry",
```

Comment:
logger.Warnf maybe?

Reply:
it's should be debug but for now will keep as info